vllm_detector_adapter/generative_detectors/granite_guardian.py

```diff
@@ -95,6 +111,107 @@ def __preprocess(
 
         return request
 
+    def _make_tools_request(
+        self, request: ChatDetectionRequest
+    ) -> Union[ChatDetectionRequest, ErrorResponse]:
+        """Convert original chat detection request to Granite Guardian-compatible request with
+        tools, user, and assistant messages."""
+
+        if (
+            "risk_name" not in request.detector_params
+            or request.detector_params["risk_name"] not in self.TOOLS_RISKS
+        ):
+            # Provide error here, since otherwise follow-on tools message
+            # and assistant message flattening will not be applicable
+            return ErrorResponse(
+                message="tools analysis is not supported with given risk",
+                type="BadRequestError",
+                code=HTTPStatus.BAD_REQUEST.value,
+            )
+
+        # 'Flatten' the assistant message, extracting the tool_calls functions
+        # Also check for user messages
+        assistant_message_idxs = []
+        assistant_message = None
+        user_message_idxs = []
+        user_message = None
+        # NOTE: Guardian models expect json to be serialized and indented
+        for i, message in enumerate(request.messages):
+            assistant_tool_call_functions = []
+            if (
+                message["role"] == "assistant"
+                and "tool_calls" in message
+                and message["tool_calls"]
+            ):
+                for tool_call in message["tool_calls"]:
+                    tool_call_function = tool_call["function"]
+                    # OpenAI stores function arguments as a str, but
+                    # Granite Guardian expects the arguments as a Dict.
+                    # We still convert so that there is no unexpected behavior
+                    # when all the functions are converted to json string
+                    # with indentation
+                    arg_dict = json.loads(tool_call_function["arguments"])
```

Comment:
since argument list can be big, am wondering if we should use more optimized json parsers, like [orjson](https://github.com/ijl/orjson)